swarm/storage/ldbstore.go

```diff
@@ -256,30 +289,27 @@ func (s *LDBStore) collectGarbage(ratio float32) {
 	defer it.Release()
 
 	garbage := []*gcItem{}
-	gcnt := 0
+	var gcnt uint64
+	maxGcnt := s.getGCCount()
 
-	for ok := it.Seek([]byte{keyIndex}); ok && (gcnt < maxGCitems) && (uint64(gcnt) < s.entryCnt); ok = it.Next() {
```

Comment:
I am not sure why this is changed. The previous condition reads correct to me. We keep on searching for items to garbage collect, until we have hit the last item (s.entryCnt) (this condition probably doesn't make much sense), or we have hit the `maxGCItems`.